src/posix/platform/netif.cpp

```diff
@@ -152,6 +152,40 @@ char         gNetifName[IFNAMSIZ];
 
 #if OPENTHREAD_CONFIG_PLATFORM_NETIF_ENABLE
 
+namespace {
+
+/**
+ * This utility class converts binary IPv6 address to text format.
+ *
+ */
+class Ip6AddressString
```

Comment:
There is already a `ToString` method in the internal `Ip6::Address` class:
https://github.com/openthread/openthread/blob/fa57212b82e34a95c6b3302ee9ec2047388ab342/src/core/net/ip6_address.hpp#L880-L886

Is it applicable at here?

Reply:
as you see, it's in the core library.